packages/invoices/src/OutboundShipment/DetailView/modals/ItemDetailsModal.tsx

```diff
@@ -88,7 +88,7 @@ const useBatchRows = (summaryItem: OutboundShipmentSummaryItem | null) => {
     setBatchRows(() => {
       return data
         .map(batch => {
-          const matchingInvoiceRow = summaryItem.batches.find(
+          const matchingInvoiceRow = Object.values(summaryItem.batches).find(
```

Comment:
I've ended up making `batches` a lookup of id:batch as it makes it a little easier to search for a specific batch (iterate through items, and then through batches), so there's a decent amount of transforming it back to an array...

Reply:
`Map` is a great idea. Wonder why that's not used more! Wonder if it is to do with serialisation .. 🤔 

The `Object` methods are pretty annoying! 

I also wondered if we could make `OutboundShipmentItem` an object with methods like `getRow` etc.. 🤔